ncm-grub/src/main/perl/grub.pm

```diff
@@ -191,11 +193,25 @@ sub password
     my $val = $tree->{option} ? "--$tree->{option} " : "";
     $val .= $password;
 
-    $grub_fh->add_or_replace_lines(qr/^password\s+/,
-                                   qr/^password $val$/,
-                                   "password $val\n",
-                                   $self->main_section_offset($grub_fh),
-                                   );
+    if ( -d $GRUB2_DIR ) {
```

Comment:
this is probably not sufficient. check `man grubby` on eg el7 and look for the location of files on certain arch/bios combos.
`/etc/grub2.cfg` is an EL7-ism that is a symlink to whatever arch/bios combo; so you might also check that (or only check that).